diff --git a/projects/SelfTest/MiscTests.cpp b/projects/SelfTest/MiscTests.cpp
index 49cfcf63..86bd4f39 100644
--- a/projects/SelfTest/MiscTests.cpp
+++ b/projects/SelfTest/MiscTests.cpp
@@ -381,8 +381,8 @@ TEST_CASE( "toString on wchar_t returns the string contents", "[toString]" ) {
 	CHECK( result == "\"wide load\"" );
 }
 
-TEST_CASE( "Divide by Zero signal handler", "[.][sig]" ) {
-    int i = 0;
-    int x = 10/i; // This should cause the signal to fire
-    CHECK( x == 0 );
-}
+//TEST_CASE( "Divide by Zero signal handler", "[.][sig]" ) {
+//    int i = 0;
+//    int x = 10/i; // This should cause the signal to fire
+//    CHECK( x == 0 );
+//}
