diff --git a/src/catch2/reporters/catch_reporter_combined_tu.cpp b/src/catch2/reporters/catch_reporter_combined_tu.cpp
index 03f8e4a3..a05b4998 100644
--- a/src/catch2/reporters/catch_reporter_combined_tu.cpp
+++ b/src/catch2/reporters/catch_reporter_combined_tu.cpp
@@ -205,10 +205,11 @@ namespace Catch {
         }
 
         if (isFiltered) {
-            out << pluralise(tests.size(), "matching test case") << "\n\n" << std::flush;
+            out << pluralise(tests.size(), "matching test case");
         } else {
-            out << pluralise(tests.size(), "test case") << "\n\n" << std::flush;
+            out << pluralise(tests.size(), "test case");
         }
+        out << "\n\n" << std::flush;
     }
 
 } // namespace Catch
