diff --git a/projects/SelfTest/ApproxTests.cpp b/projects/SelfTest/ApproxTests.cpp
index 6b7cf44a..8e7e33c2 100644
--- a/projects/SelfTest/ApproxTests.cpp
+++ b/projects/SelfTest/ApproxTests.cpp
@@ -160,7 +160,7 @@ inline std::ostream& operator<<( std::ostream& os, StrongDoubleTypedef td ) {
 TEST_CASE
 (
  "Comparison with explicitly convertible types",
- "[Approx]"
+ "[Approx][c++11]"
 )
 {
   StrongDoubleTypedef td(10.0);
