diff --git a/tests/SelfTest/UsageTests/Matchers.tests.cpp b/tests/SelfTest/UsageTests/Matchers.tests.cpp
index 93543434..cf6f3ce7 100644
--- a/tests/SelfTest/UsageTests/Matchers.tests.cpp
+++ b/tests/SelfTest/UsageTests/Matchers.tests.cpp
@@ -245,12 +245,8 @@ namespace { namespace MatchersTests {
 
             ~CustomAllocator() = default;
 
-            using std::allocator<T>::address;
             using std::allocator<T>::allocate;
-            using std::allocator<T>::construct;
             using std::allocator<T>::deallocate;
-            using std::allocator<T>::max_size;
-            using std::allocator<T>::destroy;
         };
 
         TEST_CASE("Vector matchers", "[matchers][vector]") {
