diff --git a/projects/SelfTest/ExceptionTests.cpp b/projects/SelfTest/ExceptionTests.cpp
index affac4b5..4bcfb7ac 100644
--- a/projects/SelfTest/ExceptionTests.cpp
+++ b/projects/SelfTest/ExceptionTests.cpp
@@ -73,7 +73,7 @@ TEST_CASE( "When unchecked exceptions are thrown during a REQUIRE the test shoul
     FAIL( "This should never happen" );
 }
 
-TEST_CASE( "When unchecked exceptions are thrown during a CHECK the test should continue", "[!throws]" )
+TEST_CASE( "When unchecked exceptions are thrown during a CHECK the test should continue", "[.][failing][!throws]" )
 {
     try {
         CHECK(thisThrows() == 0);
