diff --git a/src/catch2/benchmark/detail/catch_complete_invoke.cpp b/src/catch2/benchmark/detail/catch_complete_invoke.cpp
index c074f7a3..8fd43d12 100644
--- a/src/catch2/benchmark/detail/catch_complete_invoke.cpp
+++ b/src/catch2/benchmark/detail/catch_complete_invoke.cpp
@@ -8,7 +8,10 @@
 namespace Catch {
     namespace Benchmark {
         namespace Detail {
+            CATCH_INTERNAL_START_WARNINGS_SUPPRESSION
+            CATCH_INTERNAL_SUPPRESS_GLOBALS_WARNINGS
             const std::string benchmarkErrorMsg = "a benchmark failed to run successfully";
+            CATCH_INTERNAL_STOP_WARNINGS_SUPPRESSION
         } // namespace Detail
     } // namespace Benchmark
 } // namespace Catch
