diff --git a/projects/SelfTest/MiscTests.cpp b/projects/SelfTest/MiscTests.cpp
index e98b2630..02099a71 100644
--- a/projects/SelfTest/MiscTests.cpp
+++ b/projects/SelfTest/MiscTests.cpp
@@ -400,8 +400,8 @@ static int f() {
     return 1;
 }
 
-TEST_CASE( "#835 -- errno should not be touched by Catch" ) {
-  errno = 1;
-  CHECK(f() == 0);
-  REQUIRE(errno == 1); // Check that f() doesn't touch errno.    
+TEST_CASE( "#835 -- errno should not be touched by Catch", "[!shouldfail]" ) {
+    errno = 1;
+    CHECK(f() == 0);
+    REQUIRE(errno == 1); // Check that f() doesn't touch errno.    
 }
