diff --git a/include/internal/catch_output_redirect.cpp b/include/internal/catch_output_redirect.cpp
index c3ad826d..fabf9cf1 100644
--- a/include/internal/catch_output_redirect.cpp
+++ b/include/internal/catch_output_redirect.cpp
@@ -1,9 +1,9 @@
 /*
-*  Created by Martin on 28/04/2018.
-*
-*  Distributed under the Boost Software License, Version 1.0. (See accompanying
-*  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
-*/
+ *  Created by Martin on 28/04/2018.
+ *
+ *  Distributed under the Boost Software License, Version 1.0. (See accompanying
+ *  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ */
 
 #include "catch_output_redirect.h"
 
@@ -13,6 +13,7 @@
 #include <cstring>
 #include <fstream>
 #include <sstream>
+#include <stdexcept>
 
 #if defined(CATCH_PLATFORM_WINDOWS)
 #include <io.h>      //_dup and _dup2
